src/components/MultiSelect/MultiSelectMenuButton.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import Button from "./../Button/Button";
+import Icon from "./../Icons/Icon";
+import { Box, useMultiStyleConfig } from "@chakra-ui/react";
+import { SelectedItems } from "./MultiSelectTypes";
+
+export interface MultiSelectMenuButtonProps {
+  /** The id of the multiSelect using this button. */
+  multiSelectId: string;
```

Comment:
Because of the donwshift-js props passed through, the id variable is populated by the button id. So the MultiSelect's id has to have a different variable here.